framework/db/mysql/QueryBuilder.php

```diff
@@ -147,6 +147,18 @@ public function checkIntegrity($check = true, $schema = '', $table = '')
     /**
      * @inheritdoc
      */
+    public function buildFrom($tables, &$params)
+    {
+        if (count($tables) < 2) {
```

Comment:
`buildFrom` has no idea whether there is a join or not. If we want to add a check for that, this has to be done in `build()` but I do not see a good way for overriding that one.